StackExchange.Redis/StackExchange/Redis/ConnectionMultiplexer.cs

```diff
@@ -37,13 +37,16 @@ public static Task<T> ObserveErrors<T>(this Task<T> task)
             return task;
         }
 
+        [ThreadStatic]
+        public static bool ContinueOnCapturedContext;
```

Comment:
I used this technique to avoid having to re-factory the existing code too much. `SynchronizationContext.Current` is also thread-local so it seems to make sense to me. I suppose ForAwait can change to something like

```csharp
public static ConfiguredTaskAwaitable ForAwait(this Task task)
{
    var useContext = SynchronizationContext.Current.GetType() == typeof(SychronousSynchronizationContext);
    return task.ConfigureAwait(useContext);
}

Alternatively we could pass true/false to every async method.

How do you feel about using a custom SyncrhonizationContext implementation in general? I chose this method because it required the least amount of code change and allowed the use of async/await to continue.